pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -764,7 +764,7 @@ type (
 	// NodeGroupInstancesDistribution holds the configuration for spot instances
 	NodeGroupInstancesDistribution struct {
 		//+required
-		InstanceTypes []string `json:"instanceTypes,omitempty"`
+		InstanceTypes []string `json:"instanceTypes,omitempty" jsonschema:"required"`
```

Comment:
the comment is saying this is required

Reply:
I think we might forget about these. shouldn't this tag be derived from the lack of `omitempty`?

Reply:
we could use `json:omitempty` to determine if the field is actually mandatory or not. However, some places we are using pointer to avoid nil value (this one is kind of golang json encode/decoder behavior). So I think that is why in https://github.com/weaveworks/eksctl/issues/2052 @cPu1 proposed using `jsonschema:required` tag for generating json schema file. I kind of like this seperation of concern, however, we need to manage both tags.